src/Arcus.EventGrid.Proxy.Api/Startup.cs

```diff
@@ -29,37 +31,34 @@ public void ConfigureServices(IServiceCollection services)
             services.AddMvc()
                 .AddJsonOptions(jsonOptions =>
                 {
-                    jsonOptions.SerializerSettings.Converters.Add(new StringEnumConverter());
-                    jsonOptions.SerializerSettings.NullValueHandling = NullValueHandling.Ignore;
+                    jsonOptions.JsonSerializerOptions.Converters.Add(new JsonStringEnumConverter());
+                    jsonOptions.JsonSerializerOptions.IgnoreNullValues = true;
                 })
-                .SetCompatibilityVersion(CompatibilityVersion.Version_2_2);
+                .SetCompatibilityVersion(CompatibilityVersion.Version_3_0);
 
-            services.UseOpenApiSpecifications();
             services.AddSingleton(BuildEventGridPublisher);
 
-            ValidateConfiguration();
-        }
-
-        private void ValidateConfiguration()
-        {
-            var validationOutcomes = RuntimeValidator.Run(Configuration);
+            services.UseOpenApiSpecifications();
+            services.AddCorrelation();
+            services.AddHealthChecks();
 
-            if (validationOutcomes.Any(validationOutcome => validationOutcome.Successful == false))
-            {
-                throw new Exception("Unable to start up due to invalid configuration");
-            }
+            ValidateConfiguration();
         }
 
         // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
-        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
+        public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
         {
             if (env.IsDevelopment())
             {
                 app.UseDeveloperExceptionPage();
             }
-
-            app.UseMvc();
+            
+            app.UseRouting();
+            app.UseMiddleware<ExceptionHandlingMiddleware>();
```

Comment:
I think we can place the exception handling before the routing bc the handling doesn't require any routing logic.